mempool/src/mempool_service.rs

```diff
@@ -27,28 +29,21 @@ pub(crate) struct MempoolService {
     pub(crate) core_mempool: Arc<Mutex<CoreMempool>>,
 }
 
+#[tonic::async_trait]
 impl Mempool for MempoolService {
-    fn add_transaction_with_validation(
-        &mut self,
-        ctx: ::grpcio::RpcContext<'_>,
-        req: crate::proto::mempool::AddTransactionWithValidationRequest,
-        sink: ::grpcio::UnarySink<crate::proto::mempool::AddTransactionWithValidationResponse>,
-    ) {
+    async fn add_transaction_with_validation(
+        &self,
+        request: tonic::Request<AddTransactionWithValidationRequest>,
+    ) -> Result<tonic::Response<AddTransactionWithValidationResponse>, tonic::Status> {
+        let req = request.into_inner();
         trace!("[GRPC] Mempool::add_transaction_with_validation");
-        let _timer = SVC_COUNTERS.req(&ctx);
```

Comment:
Did you mean to eliminate all the SVC_COUNTERS stuff?

Reply:
The issue is that SVC_COUNTERS isn't compatible with tonic as it relies on the grpcio types that we're removing.

Reply:
Will that come back later somehow with tonic types? Did you talk with the telemetry folks to make sure these aren't going to be missed?

Reply:
@sherry-x You're leading the charge on metrics, do you have any thoughts on this? Should we try to incorporate a SVC_COUNTERS that supports tonic types before landing these?

Reply:
Do we need those counters, are they used anywhere? If not we should just remove them all and cleanup ServiceMetrics later, if we still need them, we can rewrite it in the new format.

Reply:
@sherry-x I don't know if they are used, that was why I was asking you :D Sounds like its just some legacy counters then?

Reply:
Looks like it :D